id_table.h

```diff
@@ -15,12 +15,30 @@ enum rb_id_table_iterator_result {
     ID_TABLE_ITERATOR_RESULT_END
 };
 
+typedef struct rb_id_item item_t;
```

Comment:
`item_t` should not be exposed, IMO.

Reply:
I moved the typedef to `id_table.c`. Let me know if you prefer having it as a void pointer instead of `struct rb_id_item *`.